src/main/java/com/marklogic/kafka/connect/source/JsonRowBatcherSupplier.java

```diff
@@ -0,0 +1,59 @@
+package com.marklogic.kafka.connect.source;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.marklogic.client.datamovement.DataMovementManager;
+import com.marklogic.client.datamovement.RowBatchSuccessListener;
+import com.marklogic.client.datamovement.RowBatcher;
+import com.marklogic.client.io.Format;
+import com.marklogic.client.io.JacksonHandle;
+import com.marklogic.client.io.marker.ContentHandle;
+import org.apache.kafka.connect.source.SourceRecord;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.Map;
+
+public class JsonRowBatcherSupplier extends RowBatcherSupplier {
+    protected final Logger logger = LoggerFactory.getLogger(getClass());
+    private final String topic;
+    private final DataMovementManager dataMovementManager;
+    private final Map<String, Object> parsedConfig;
+    private final List<SourceRecord> sourceRecords;
+
+    JsonRowBatcherSupplier(DataMovementManager dataMovementManager, List<SourceRecord> sourceRecords, Map<String, Object> parsedConfig) {
+        this.dataMovementManager = dataMovementManager;
+        this.sourceRecords = sourceRecords;
+        this.parsedConfig = parsedConfig;
+        this.topic = (String) parsedConfig.get(MarkLogicSourceConfig.TOPIC);
+    }
+
+    public RowBatcher<JsonNode> get() {
```

Comment:
Seeing this, I think I gave bad advice with recommending the use of Supplier. This of course works fine, but I don't like how `get()` is ambiguous about whether it returns a new instance or the same instance - the javadocs for Supplier even state that it's intended to be ambiguous. And because `get()` is no-arg, it means that the supplier has to hold onto the List object as state. 

I think a `Function<List<SourceRecord>, RowBatcher>` is the better way to go - but that's a bit ugly, and the Function method of `apply` is not intuitive either. So I think a custom interface is better, like you had in mind originally:

```
interface RowBatcherBuilder 
  RowBatcher newRowBatcher(List<SourceRecord> list)
```

That has another benefit too - you can now look at output type in the `start()` method and instantiate a `RowBatcherBuilder` and reuse it on every call to `poll()`. The RBB will need the DataMovementManager and config Map passed to its constructor, and it can hold onto both of those as class fields because both are not intended to be modified. 

Then, `poll()` just does the following:

```
    List<SourceRecord> records = new ...
    RowBatcher rowBatcher = this.rowBatcherBuilder.newRowBatcher(records);
```

That makes it clear that a new list is created every time and it's used to create the RowBatcher, and thus there's no reason for the RBB to hold onto it as state. 

Sorry for the bad advice with Supplier, but I think this is a worthwhile refactor.